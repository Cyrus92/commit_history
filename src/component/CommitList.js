import React from 'react';
import Button from 'react-bootstrap/Button';
import CommitRow from './CommitRow.js';

export default class CommitList extends React.Component {
    constructor(props) {
        super(props);
    }


  render() {
    return (
        <div>
            <br /><br />
            {/* ROWS */}            
            {
                (Array.isArray(this.props.commitsData) && this.props.commitsData.length)?
                this.props.commitsData.map( function(commitItem,i){
                    return <CommitRow key={i} githubCommitItem={commitItem} />             
                            
                })
                :"No available data"

            }
            <br />
            {/* LOAD MORE BUTTON */}
            {
                (Array.isArray(this.props.commitsData) && this.props.commitsData.length)?
                <Button id="loadMoreBtn" variant="primary" >More (...) </Button>
                :""
            }
            <br /><br />
            
            
        </div>
    ) 
  }
}