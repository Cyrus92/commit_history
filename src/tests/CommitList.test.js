import React from 'react';
import { shallow, mount } from 'enzyme';
import CommitList from '../component/CommitList';
describe("CommitList component", () => {

    it("should render", () => {
        const wrapper = shallow(<CommitList />);
    });

});