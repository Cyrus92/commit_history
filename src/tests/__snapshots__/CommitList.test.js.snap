// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommitList component should render initial layout 1`] = `
Array [
  <div>
    <br />
    <br />
    No available data
    <br />
    
    <br />
    <br />
  </div>,
]
`;
